File: ios/Runner/NativeButton.swift
import Flutter
import UIKit

/// Platform view that displays a native UIButton and sends tap events to Flutter.
final class NativeButton: NSObject, FlutterPlatformView {
  // MARK: - Properties

  private let button: UIButton
  private let channel: FlutterMethodChannel

  // MARK: - Initialization

  init(
    frame: CGRect,
    viewIdentifier viewId: Int64,
    arguments args: Any?,
    binaryMessenger messenger: FlutterBinaryMessenger
  ) {
    self.button = UIButton(type: .system)
    self.channel = FlutterMethodChannel(
      name: "com.masahikokobayashi/nativeButton",
      binaryMessenger: messenger
    )
    super.init()

    button.setTitle("Native Button", for: .normal)
    button.addTarget(self, action: #selector(handleTap), for: .touchUpInside)
  }

  // MARK: - FlutterPlatformView

  func view() -> UIView {
    return button
  }

  // MARK: - Actions

  @objc private func handleTap() {
    channel.invokeMethod("onTap", arguments: nil)
  }
}

File: android/app/src/main/kotlin/com/masahikokobayashi/flutterlab/infrastructure/platform/NativeButton.kt
package com.masahikokobayashi.flutterlab.infrastructure.platform

import android.content.Context
import android.view.View
import android.widget.Button
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.platform.PlatformView

/// Platform view that displays a native Button and sends tap events to Flutter.
class NativeButton(
    context: Context,
    viewId: Int,
    args: Any?,
    messenger: BinaryMessenger,
) : PlatformView {
    private val button: Button = Button(context)
    private val channel: MethodChannel =
        MethodChannel(messenger, "com.masahikokobayashi/nativeButton")

    init {
        button.text = "Native Button"
        button.setOnClickListener { handleTap() }
    }

    override fun getView(): View = button

    override fun dispose() {}

    /// Sends a tap event to Flutter via the method channel.
    private fun handleTap() {
        channel.invokeMethod("onTap", null)
    }
}
